packages/medusa/src/api-v2/admin/promotions/route.ts

```diff
@@ -21,3 +22,16 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
     limit,
   })
 }
+
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  const createPromotions = createPromotionsWorkflow(req.scope)
+  const manager = req.scope.resolve("manager")
+  const promotionsData = [req.validatedBody as CreatePromotionDTO]
+
+  const { result } = await createPromotions.run({
+    input: { promotionsData },
+    context: { manager },
+  })
```

Comment:
todo: we should handle the errors the same way everywhere no? we need to settle on something. I know that in the create product etc we do throw on error and we don't handle it manually

Reply:
correct, missed this one.